src/services/item/repository.ts

```diff
@@ -555,6 +557,33 @@ export class ItemRepository {
     }
   }
 
+  /**
+   * Return a copy with a suffix of the string given in parameter. The suffix is determined by the siblings of the item.
+   * If there is no sibling with the same name, the copied name will be the same as the original name.
+   * The suffix respect the format " (0)". "0" is a succint positive number starting at 2.
+   * If the string given in parameter already have a valid suffix, increment the number.
+   * If the copied name exceed the maximum characters allowed, the original name will be shorten from the end,
+   * the copied name will be equals to the maximum allowed.
+   * @param name string to copy.
+   * @param siblings list of siblings to be used to check if the copied name already exist.
+   * @returns a copy of the string given in parameter, with a suffix.
+   */
+  private addCopySuffix(name: string, siblings: Item[]): string {
+    const maxIterations = siblings.length;
+    let result = name;
+    for (let i = 0; i < maxIterations; i++) {
+      const twinIndex = siblings.findIndex((sibling) => sibling.name === result);
+      if (twinIndex >= 0) {
+        result = this.incrementCopySuffix(result);
+        // By definition, the result string will never be the same as the current twin, so we can remove it from the list.
+        siblings.splice(twinIndex, 1);
```

Comment:
This is a side effect of the `addCopySuffix` and its not specified in the JSDoc. If possible, we should avoid to update the state of an input parameter, because it can lead to unexpected behaviour. If you have to update the array of siblings (I didn't check the whole code now), it could be good to indicate in the JSDoc that the siblings will be mutated.

Reply:
Good catch, I'm not expecting this function to have an impact on the parameter. I'll just copy the array to an another one.